### PR TITLE
Add encoding.tison - a faster version of encoding.ubjson

### PR DESCRIPTION
**Pull request by kasperl**

### CODE REVIEW

File: tests/tison_test.toit [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+test_maps -> none:
+  test_round_trip {:}
+  test_round_trip {"foo": 42}
+  test_round_trip {"bar": {"baz": null}}
+
+test_lists -> none:
+  test_round_trip [0, 1, 2, 3]
+  test_round_trip [1, 2, 3, 4]
+  // TODO(kasper): enconding.tison deals incorrectly with large
```
Comment (erikcorry) [inline tests/tison_test.toit:56]:
```suggestion
  // TODO(kasper): encoding.tison deals incorrectly with large
```

**Review by erikcorry — APPROVED** [review]

File: src/primitive_encoding.cc [hunk 2]
```diff
@@ -194,4 +190,43 @@ PRIMITIVE(hex_decode)  {
...
+  ARGS(Object, object);
+
+  int length = 0;
+  { MessageEncoder size_encoder(process, null, true);
+    if (!size_encoder.encode(object)) WRONG_TYPE;
+    length = size_encoder.size();
+  }
+
+  ByteArray* result = null;
+  if (length <= ByteArray::max_internal_size_in_process()) {
```
[the remark this answers is not in the document]
Comment (kasperl) [reply src/primitive_encoding.cc]: Done. Thanks!

File: src/process.h [hunk 3]
```diff
@@ -150,14 +150,13 @@ class Process : public ProcessListFromProcessGroup::Element,
   void set_current_directory(int fd) { _current_directory = fd; }
   int gc_count(GcType type) { return _object_heap.gc_count(type); }
 
-  // Special allocation of byte arrays and strings due to multiple reasons for failure.
-  // The error string is only set if null is returned.
-  String* allocate_string(const char* content, Error** error);
-  String* allocate_string(int length, Error** error);
-  String* allocate_string(const char* content, int length, Error** error);
+  // Special allocation of byte arrays and strings.
```
[the remark this answers is not in the document]
Comment (kasperl) [reply src/process.h:153]: Removed.

File: tests/tison_test.toit (hunk 1 above)
Comment (kasperl) [reply tests/tison_test.toit:56]: Fixed.

**Pull request merged**